java/code/src/com/suse/manager/utils/SaltUtils.java

```diff
@@ -2029,28 +2028,31 @@ public static String uuidToLittleEndian(String uuidIn) {
         return uuidSwap.toString().replaceAll("-", "");
     }
 
-    /**
-     * Decode the std message from the whole message
-     *
-     * @param message the message Object
-     * @param key the json key of the message to decode (e.g.: sdterr, stdout)
-     * @return the String decoded if it exists
-     */
-    public static String decodeStdMessage(Object message, String key) {
-        JsonElement json = null;
-        if (message instanceof JsonParsingError) {
-            json = ((JsonParsingError)message).getJson();
-        }
-        else if (message instanceof SaltSSHError) {
-            try {
-                // Try parsing the error message as JSON
-                json = JsonParser.GSON.fromJson("{" + ((SaltSSHError) message).getMessage() + "}", JsonElement.class);
-            }
-            catch (JsonParseException ex) {
-                LOG.debug("Unable to parse SaltSSHError message as json");
+    private static JsonElement parseJsonError(SaltError message) {
+        return message.fold(
+            functionNotAvailable -> null,
+            moduleNotSupported -> null,
+            jsonParsingError -> jsonParsingError.getJson(),
+            genericError -> null,
+            saltSshError -> {
+                try {
+                    // Try parsing the error message as JSON
+                    return JsonParser.GSON.fromJson("{" + saltSshError.getMessage() + "}", JsonElement.class);
+                }
+                catch (JsonParseException ex) {
+                    LOG.info("Unable to parse SaltSSHError message \"{}\"as json: {}",
```

Comment:
I think I agree with @mcalmer here to make this a `WARN` entry.
We should be able to parse all the expected messages. If we cannot parse something, it's something unexpected and it needs to be addressed by improving our error handling.